File: Solutions/List/FindPeakElement.py
def findPeakElement(nums):
    for i in range(0, len(nums), 2):  
        if i == len(nums) - 1:
            if nums[i-1] < nums[i]:
                return i
        if i == 0:
            if len(nums) == 1:
                return 0
            if nums[i+1] < nums[i]:
                return i
        elif nums[i] < nums[i-1] > nums[i-2]:
            print("A")
            return i - 1
        else:
            if nums[i - 1] < nums[i] > nums[i + 1]:
                return i
    return len(nums) - 1


print(findPeakElement([1,2,1]))

File: FindPeakElement.ts
function findPeakElement(nums: number[]): number {
    let l: number = 0;
    let r: number = nums.length - 1;

    while (l < r) {
        let mid: number = Math.floor((l + r) / 2);

        if (nums[mid] > nums[mid + 1]) {
            r = mid
        } else {
            l = mid + 1
        }
    }

    return l;
};